### DISCUSSION TITLE
StartAsyncSerialServer doesn't work pymodbus version 3.2???

### DISCUSSION THREAD

**Discussion by dlmoffett** [pymodbus-dev/pymodbus#1433] [Help] [answered]
Before I create a bug ticket, I thought I better ask the community about this one 'cause maybe I'm just doing something reeeeeal dumb?

### Details
Running python natively on Windows; no WSL/docker/etc....

From my poetry lock file, here are the exact details of the version of pymodbus I'm using:
```
[[package]]
name = "pymodbus"
version = "3.2.0"
description = "A fully featured modbus protocol stack in python"
category = "main"
optional = false
python-versions = ">=3.8.0"
files = [
    {file = "pymodbus-3.2.0-py2.py3-none-any.whl", hash = "sha256:ca954cf0c7e88e7cfc0d273960919b68090e72e2a88f39d2a3b50f4ab8eaf2a9"},
    {file = "pymodbus-3.2.0.tar.gz", hash = "sha256:5769ac9a00ffc154e1761c69299c1968dcab1c2a61741b1caff89c4aea81dbcf"},
]
```

### Sanity Code
```python
import asyncio
import logging

from pymodbus.framer.rtu_framer import ModbusRtuFramer
from pymodbus.server import StartAsyncSerialServer

logging.basicConfig(
    format="[%(asctime)s] [%(levelname)s] [%(name)s] %(message)s",
    datefmt="%Y-%m-%d %H:%M:%S %z",
    level=logging.INFO,
)
log = logging.getLogger(__name__)


async def modbus_slave():
    return await StartAsyncSerialServer(
        framer=ModbusRtuFramer,
        port="COM10",
        baudrate=19200,
    )


def sanity():
    log.info(f"✅ Here I am!")
    asyncio.run(modbus_slave())
    log.info(f"❌ I'm going insane!")
```

Per the [example documentation](https://pymodbus.readthedocs.io/en/dev/source/examples.html#asynchronous-server-example) I would expect to only see my first log statement and for the server to be up and running indefinitely until I send a keyboard interrupt. 

However, that's not the case and I see the following output when running the above code, which exits immediately after making the logs:
```
[2023-03-13 16:07:53 -0600] [INFO] [fix_pymodbus.sanity] ✅ Here I am!
[2023-03-13 16:07:53 -0600] [INFO] [fix_pymodbus.sanity] ❌ I'm going insane!
```

### Working Monkey Patch
If I make the following monkey patch everything works as expected:

```python
async def StartAsyncSerialServer(  # pylint: disable=invalid-name,dangerous-default-value
    context=None,
    identity=None,
    custom_functions=[],
    **kwargs,
):  # pragma: no cover
    """Start and run a serial modbus server.

    :param context: The ModbusServerContext datastore
    :param identity: An optional identify structure
    :param custom_functions: An optional list of custom function classes
        supported by server instance.
    :param kwargs: The rest
    """
    server = ModbusSerialServer(
        context, kwargs.pop("framer", ModbusAsciiFramer), identity=identity, **kwargs
    )
    await server.start()    # <----------------------Adding this makes it work 🤔
    await _serverList.run(server, custom_functions)
```

### My questions
1. Am I just doing something obviously wrong when running on Windows?
2. Is the example documentation out of date or am I looking at wrong documentation - is there a new way of starting a serial server?
3. Can anyone confirm that they can start a serial server on windows which runs correctly and responds to commands?

**Comment by janiversen** [comment, marked as answer]
Seems you identified a problem, feel free to make a pull request.

I am not sure when/why the start call decided to drop out.

**Reply by dlmoffett** [reply]
Ok. Library is broken. Got it! 🤣 

I'm already trying to work on a fix for this separate issue: https://github.com/pymodbus-dev/pymodbus/issues/674

So I will convert this thread to a new issue since they're technically unrelated.

**Comment by dlmoffett** [comment]
Issue created: https://github.com/pymodbus-dev/pymodbus/issues/1434
